pkg/network/ebpf/c/protocols/tls/go-tls-location.h

```diff
@@ -101,17 +104,18 @@ static __always_inline int read_register(struct pt_regs* ctx, int64_t regnum, vo
         *(u64*)dest = tmp;
         return 0;
     #elif defined(__TARGET_ARCH_arm64)
-        // TODO Support ARM
-        /*if (regnum >= 0 && regnum < sizeof(ctx->regs)) {
-            // Verifier won't allow direct access to regs array if the index is not const
-            switch (regnum) {
-                case 0:
-                    __builtin_memcpy(dest, &ctx->regs[0], sizeof(ctx->regs[0]));
-                default:
-                    return 1;
+        volatile u64 tmp = 0;
+    #pragma unroll
+        for (int i = 0; i < NUM_REGISTERS; i++) {
+            if (i == regnum) {
+                tmp = ctx->regs[i];
             }
+        }
```

Comment:
nit: slight runtime optimization
```suggestion
        for (int i = 0; i < NUM_REGISTERS; i++) {
            if (i == regnum) {
                tmp = ctx->regs[i];
                break;
            }
        }
```

Reply:
surprisingly adding the `break` statement is triggering a verifier error somewhat similar to the one I described below
this is how the code looks like for reading a given register _without_ the `break` statement:
```
; tmp = ctx->regs[i];
 197: (79) r1 = *(u64 *)(r7 +16)
 ```
 `r7` is a context pointer. as you can see the offset is constant (16).
 
 Now with the `break` statement clang is generating this instead:
 ```
; tmp = ctx->regs[i];
185: (67) r1 <<= 3
186: (0f) r7 += r1
last_idx 186 first_idx 180
regs=2 stack=0 before 185: (67) r1 <<= 3
regs=2 stack=0 before 184: (7b) *(u64 *)(r10 -320) = r2
regs=2 stack=0 before 183: (b7) r2 = 0
regs=2 stack=0 before 182: (25) if r1 > 0x1e goto pc+177
regs=2 stack=0 before 181: (79) r1 = *(u64 *)(r8 +344)
187: (79) r1 = *(u64 *)(r7 +0)
variable ctx access var_off=(0x0; 0xf8) disallowed
```
as you can see the `r7` register holding the ctx pointer is getting incremented by a variable offset coming from `r1` which again triggers a
```
variable ctx access var_off=(0x0; 0xf8) disallowed
```

I can see if I can get around this by using some of the tricks we know, but IMO this is small potatoes so should be OK to leave as it is as well

Reply:
i'm fine with removing the break, just mention that in a comment in the code, so we'll know to not add break there in the  future